src/pool/srv_pool_map.c

```diff
@@ -126,9 +126,13 @@ update_one_tgt(struct pool_map *map, struct pool_target *target,
 			else
 				D_INFO(DF_TARGET " start reintegration.\n",
 				       DP_TARGET(target));
-			D_DEBUG(DB_MD, "change rank %u to UP\n",
-				dom->do_comp.co_rank);
-			dom->do_comp.co_status = PO_COMP_ST_UP;
+			if (dom->do_comp.co_status == PO_COMP_ST_DOWNOUT ||
+			    dom->do_comp.co_status == PO_COMP_ST_DOWN) {
+				dom->do_comp.co_status = PO_COMP_ST_UP;
+				dom->do_comp.co_in_ver = target->ta_comp.co_in_ver;
+				D_DEBUG(DB_MD, "change rank %u to UP ver %u\n",
+					dom->do_comp.co_rank, dom->do_comp.co_in_ver);
+			}
 			dom->do_comp.co_flags = 0;
```

Comment:
I think the intention here is to change the domain (node) status to UP, once if there are any targets are UP due to the reintegration.

>  It seems strange that we clear all flags even if dom is not DOWNOUT or DOWN.

I think this is "wrong", we should only change it UP if it is down and downcut, or I miss your question.

Reply:
Let me rephrase the question. For

    132 136 dom->do_comp.co_flags = 0;

do we really want to clear all flags in dom->do_comp.co_flags even if dom->do_comp.co_status is not DOWNOUT or DOWN?

Reply:
looks should set dom->do_comp.co_flags at same time-point as do_comp.co_status

Reply:
Hmm, I am not so sure about co_flags. But for now, it seems ok to set it to 0 in this case. 

Since domain status will not be DOWNOUT anymore, once there are any its targets become UP, then its co_flags can be set to 0, since DOWN2OUT only used for DOWNOUT target/domain.  @liuxuezhao ?

Reply:
>looks should set dom->do_comp.co_flags at same time-point as do_comp.co_status

will this cause issue if it only exclude partial targets from multiple ranks, then add those targets back, co_flags will not be set to 0 then?

Reply:
now the PO_COMPF_DOWN2OUT flag only checked when its status is PO_COMP_ST_DOWNOUT (to differenciate it is exlude or drain cause DOWNOUT), so seems the co_flags maybe can set together with co_status.

anyway, currently that is only used for target component, not used for domain type now. So either way should not cause logic error.